clang/lib/Sema/SemaTemplateInstantiate.cpp

```diff
@@ -1722,6 +1722,32 @@ namespace {
       return inherited::TransformLambdaBody(E, Body);
     }
 
+    ExprResult RebuildSizeOfPackExpr(SourceLocation OperatorLoc,
+                                     NamedDecl *Pack, SourceLocation PackLoc,
+                                     SourceLocation RParenLoc,
+                                     std::optional<unsigned> Length,
+                                     ArrayRef<TemplateArgument> PartialArgs) {
+      if (SemaRef.CodeSynthesisContexts.back().Kind !=
+          Sema::CodeSynthesisContext::ConstraintNormalization)
+        return inherited::RebuildSizeOfPackExpr(OperatorLoc, Pack, PackLoc,
+                                                RParenLoc, Length, PartialArgs);
+
+#ifndef NDEBUG
+      for (auto *Iter = TemplateArgs.begin(); Iter != TemplateArgs.end();
+           ++Iter)
+        for (const TemplateArgument &TA : Iter->Args)
+          assert(TA.getKind() != TemplateArgument::Pack || TA.pack_size() == 1);
+#endif
+      Sema::ArgumentPackSubstitutionIndexRAII SubstIndex(SemaRef, 0);
```

Comment:
```suggestion
      Sema::ArgumentPackSubstitutionIndexRAII SubstIndex(SemaRef, /*NewSubstitutionIndex=*/0);
```
Though why is the Substitution index always 0 here?

Reply:
That is used for transforming TemplateTemplateParmDecls e.g. in the following case:
```cpp
template <template <typename> class... TTp>
    requires(sizeof...(TTp) > 0)
  friend class TTP;
```
The TTP here is a ParameterPack, and our transformation assumes a non-negative ArgumentPackSubstitutionIndex. Fortunately, we can always expand it to the first element during the normalization because we would have only one (injected) template argument for the pack.